scio-test/src/main/scala/com/spotify/scio/testing/CoderAssertions.scala

```diff
@@ -30,102 +31,148 @@ import scala.reflect.ClassTag
 object CoderAssertions {
   private lazy val DefaultPipelineOptions = PipelineOptionsFactory.create()
 
-  implicit class CoderAssertionsImplicits[T](private val value: T) extends AnyVal {
+  type CoderAssertionT[T] = AssertionContext[T] => Assertion
+  type CoderAssertion = AssertionContextBase => Assertion
+
+  implicit class ValueShouldSyntax[T](value: T) {
     def coderShould(
-      coderAssertion: CoderAssertion[T]
-    )(implicit c: Coder[T], eq: Equality[T]): Assertion =
-      coderAssertion.assert(value)
-  }
+      coderAssertion: CoderAssertionT[T]
```

Comment:
we can accept both `CoderAssertion` and `CoderAssertionT[T]` due to contra-variance